packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -316,6 +317,38 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
       const abortController = new AbortController();
 
       if (isOnline) {
+        if (visitToEdit?.uuid) {
+          const existingAttributes = visitToEdit.attributes.map((attr) => attr.attributeType.uuid) || [];
+          const updatedAttributes = [];
+          const addedAttributes = [];
+          payload.attributes.forEach((attribute) => {
+            if (existingAttributes.includes(attribute.attributeType)) {
+              const attributeUuid = visitToEdit.attributes.filter(
+                (attr) => attr.attributeType.uuid === attribute.attributeType,
+              )[0].uuid;
+              updatedAttributes.push({ uuid: attributeUuid, value: attribute.value });
+            } else addedAttributes.push(attribute);
```

Comment:
There are so many things that can fail here:
1. If attributes is undefined
2. If filter returns empty
3. If the request to update the visit attribute fails, this is assuming a very happy case, which is not always the case.

Reply:
> There are so many things that can fail here:
> 
> 1. If attributes is undefined
> 2. If filter returns empty
> 3. If the request to update the visit attribute fails, this is assuming a very happy case, which is not always the case.

- Upon failure, I added have a snackbar error feedback.